libwasmvm/src/cache.rs

```diff
@@ -384,6 +385,64 @@ fn do_get_metrics(cache: &mut Cache<GoApi, GoStorage, GoQuerier>) -> Result<Metr
     Ok(cache.metrics().into())
 }
 
+#[derive(Serialize)]
+struct PerModuleMetrics {
+    hits: u32,
+    size: usize,
+}
+
+impl From<cosmwasm_vm::PerModuleMetrics> for PerModuleMetrics {
+    fn from(value: cosmwasm_vm::PerModuleMetrics) -> Self {
+        Self {
+            hits: value.hits,
+            size: value.size,
+        }
+    }
+}
+
+#[derive(Serialize)]
+struct PinnedMetrics {
+    per_module: HashMap<Checksum, PerModuleMetrics>,
+}
+
+impl From<cosmwasm_vm::PinnedMetrics> for PinnedMetrics {
+    fn from(value: cosmwasm_vm::PinnedMetrics) -> Self {
+        Self {
+            per_module: value
+                .per_module
+                .iter()
+                .cloned()
```

Comment:
Can't we use into_iter and only clone on demand in the map?

Reply:
Used `.into_iter()`. Tbh, I'm not quite sure why I ever just called `.iter()` instead of `.into_iter()`